pyvista/core/composite.py

```diff
@@ -1236,6 +1237,10 @@ def _activate_plotting_scalars(self, scalars_name, preference, component, rgb):
         if rgb:
             if scalars.ndim != 2 or scalars.shape[1] not in (3, 4):
                 raise ValueError('RGB array must be n_points/n_cells by 3/4 in shape.')
+            if dtype != np.uint8:
+                # uint8 is required by the mapper to display correctly
+                _validation.check_subdtype(scalars, (np.floating, np.integer), name='rgb scalars')
+                scalars_name = self._convert_to_uint8_rgb_scalars(data_attr, scalars_name)
```

Comment:
Sorry after I approved it. I was just wondering, is there any vtk documentation or discussion on the basis that uint_8 is required as a data type?

Reply:
The pyvista mappers use `color_mode='direct'` for plotting rgb scalars:
https://github.com/pyvista/pyvista/blob/693569ad478d4905ab28e5c3dc08001b784ce660/pyvista/plotting/mapper.py#L186-L188
The docs for [vtkMapper.SetColorModeToDirectScalars()](https://vtk.org/doc/nightly/html/classvtkMapper.html#a6e34c4582491a60b4621f018ff6416d8) says that float and int types should just work, and uint8 is not specifically required.

This is consistent with how `pv.DataSetMapper` / `vtkDataSetMapper` works. But  #6012 shows it's not working for `pv.CompositePolyDataMapper` / `vtk.vtkCompositePolyDataMapper`. Not sure exactly why that is though.

Reply:
Thank you. I will report this to the vtk project later. If there are no other comments, I will merge it in 24 hours.